designs/credentials/credentials-management.md

```diff
@@ -0,0 +1,65 @@
+# Credentials Management
+
+This outlines how the Toolkit produces and obtains credentials.
+
+## Terminology
+
+### Credentials
+
+Credentials allow the Toolkit to interact with AWS on a user's behalf. Service clients require a [Credentials](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Credentials.html) object, and the Toolkit obtains these objects through Credentials Providers.
+
+### Credentials Provider
+
+Credentials Providers are how the toolkit abstracts away different ways of obtaining a user's credentials, and produces Credentials objects. For example, a Shared Credentials Provider knows how to obtain credentials for a specific profile within [Shared Credentials Files](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-shared.html).
+
+A Credentials Provider is produced by a Credentials Provider Factory
+
+### Credentials Provider Factory
+
+A factory is capable of producing one or more Credentials Providers for a single credentials type. For example, a Shared Credentials related factory produces Credentials Providers for each profile found within shared credentials files.
+
+### Credentials Provider Id
+
+All Credentials Providers are uniquely identified by a Credentials Provider Id. These take the format `(credentials type):(credentials type id)`. For example, Shared Credentials Providers have the type `profile`, and the id represents the profile name. So if shared credentials contained a profile named "foo", its corresponding Credentials Provider Id would be `profile:foo`.
+
+Credentials Provider Id may be surfaced to users, however it is an internal identification construct.
```

Comment:
nit: "may be" -> is surfaced

Reply:
It is currently surfaced, but that may not always be the case